onnxruntime/core/optimizer/transpose_optimization/ort_optimizer_api_impl.cc

```diff
@@ -289,10 +289,11 @@ std::vector<uint8_t> ApiTensor::Data() const {
   auto tensor_shape_dims = utils::GetTensorShapeFromTensorProto(tensor_proto_);
   TensorShape tensor_shape{std::move(tensor_shape_dims)};
   onnxruntime::Tensor tensor(tensor_dtype, tensor_shape, cpu_allocator_);
-  ORT_THROW_IF_ERROR(utils::TensorProtoToTensor(Env::Default(), model_path_.ToPathString().c_str(),
+  ORT_THROW_IF_ERROR(utils::TensorProtoToTensor(Env::Default(), model_path_,
                                                 tensor_proto_, tensor));
   size_t num_bytes = gsl::narrow_cast<size_t>(tensor.SizeInBytes());
   const uint8_t* data = static_cast<const uint8_t*>(tensor.DataRaw());
+  // TODO: the returned data is unaligned
```

Comment:
Because we are dealing with model weights. The data will be used with mlas, which has alignment requirements.  So we cannot use std::vector storing such data. Here we need to either make the data aligned, or allocate a new buffer and copy the data out.